homeworks/08-frp/src/canvas.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  BRIKE_HEIGHT,
+  GAP,
+  CANVAS_HEIGHT,
+  CANVAS_WIDTH,
+  WINDOW_SIZE,
+} from "./constant";
+import { Point } from "./point";
+
+export class GameCanvas {
+
+  public get canvas(): HTMLCanvasElement {
+    return this._canvas;
+  }
+
+  private image = new Image();
+  private _canvas: HTMLCanvasElement;
+  private context: CanvasRenderingContext2D | null;
+
+  constructor(
+    private windowsPostitions: Point[],
+    private gamePicture: string
+  ) {
+    this.image = new Image();
+    this.image.src = this.gamePicture;
+    this._canvas = <HTMLCanvasElement>document.getElementById("canvas");
+    this.context = this.canvas.getContext("2d");
+  }
+
+  public drawGameField(): void {
+    this.canvas.width = CANVAS_WIDTH;
+    this.canvas.height = CANVAS_HEIGHT;
+    if (this.context) {
+      this.drawBrikes(this.context);
```

Comment:
No need to pass class property inside a private function -- just use it directly. 
Also make sense to check canvas context availability once.